site/en/codelabs/telink-openthread-hardware/index.lab.md

```diff
@@ -0,0 +1,739 @@
+It is connected to the internet via Ethernet and will be configured as the host for the OpenThread Border Router (OTBR).
+
+### Network Connection
+
+An internet-connected switch (or router) and several Ethernet cables.
+They are used to connect the Raspberry Pi to the Linux machine, facilitating user configuration of the Raspberry Pi via the host.
+
+### LinuxBDT
+
+Telink [Burning and Debugging Tool (BDT)](http://wiki.telink-semi.cn/wiki/IDE-and-Tools/Burning-and-Debugging-Tools-for-all-Series/), which applies to all Telink Chip series, allows you to erase and flash OpenThread firmware onto the Telink TLSR9518 Development Kits.
+Please install the X86-based linux version [linuxBDT](http://wiki.telink-semi.cn/tools_and_sdk/Tools/BDT/LinuxBDT.tar.bz2) on your Linux machine.
+
+### Others
+
+* Git, for setting up Telink Zephyr Development Environment.
+* West, for manage Zephyr project and build OpenThread binaries.
+
+## Set up Firmware
+
+Duration: 12:00
+
+### Set up Telink Zephyr Development Environment
+
+On the Linux machine, open an CLI terminal and start by executing the following commands to ensure that your APT is up-to-date.
+
+```bash
+$ sudo apt update
+$ sudo apt upgrade
+```
+
+ Once that is done, proceed with the following steps.
+
+1. Install the dependencies
+
+     ```bash
+     $ wget https://apt.kitware.com/kitware-archive.sh
+     $ sudo bash kitware-archive.sh
+     $ sudo apt install --no-install-recommends git cmake ninja-build \
+     gperf ccache dfu-util device-tree-compiler python3-dev python3-pip \
+     python3-setuptools python3-tk python3-wheel xz-utils file make gcc \
+     gcc-multilib g++-multilib libsdl2-dev
+     ```
+
+     Zephyr currently requires minimum versions of main dependencies, such as CMake (3.20.0), Python3 (3.6), and Devicetree Compiler (1.4.6).
+
+     ```bash
+     $ cmake --version
+     $ python3 --version
+     $ dtc --version
+     ```
+
+     Verify the installed versions on your system before proceeding with the next steps.
+     If the versions are not correct, switch the APT mirror to a stable and up-to-date mirror, or manually update these dependencies.
+
+2. Install west
+
+     ```bash
+     $ pip3 install --user -U west
+     $ echo 'export PATH=~/.local/bin:"$PATH"' >> ~/.bashrc
+     $ source ~/.bashrc
+     ```
+
+     Make sure that `~/.local/bin` is in your `$PATH` environment variable.
+
+3. Get the Zephyr Project source code
+
+     ```bash
+     $ west init ~/zephyrproject
+     $ cd ~/zephyrproject
+     $ west update
+     $ west blobs fetch hal_telink
+     $ west zephyr-export
+     ```
+
+4. Install additional Python dependencies for Zephyr
+
+     ```bash
+     $ pip3 install --user -r ~/zephyrproject/zephyr/scripts/requirements.txt
+     ```
+
+5. Set up the Zephyr toolchain
+
+     Download the Zephyr toolchain (about 1~2 GB) to a local directory to allow you to flash most boards.
```

Comment:
Our Markdown parser for codelabs doesn't handle this well for lists.  Can you combine these into a single paragraph (also add period at the end of the "Set up" sentence).  So like

`5. Set up the Zephyr toolchain. Download the Zephyr toolchain...`

Reply:
Added in the updated commit